### ISSUE TITLE
DUA year permissions

### ISSUE THREAD

**Issue by stratabuild** [stratasan/wrike-test#104]
**The Problem we are trying to solve** 
Client DUA Permissions are having to be ﹍manually﹍ updated when a new year of data is received. It would be "nice" if these permissions were automatically applied when a new year is received to all applicable users (assuming the users org has sent their data to Stratasan). In addition when a data set is updated ALL clients with access to that dataset are receiving emails that the data has been updated regardless if they have sent their data to Stratasan or not. (_Ex. HCA sends Kentucky data. Data is updated and ALL clients with access to Kentucky data receive an email that it's been updated when in reality only HCA should have access and receive an email. Other clients will gain access once they send in the data and we (Mac) verifies its the same data set)._ This causes clients confusion and inevitably leads to clients contacting CS etc. 

**More about the problem**

**Risks** 
_From Brian_: On this we really want to weigh the risk of _unintentionally sharing data where we shouldn't_. The reason that this process is onerous is that we want a manual sign off and a directly responsible individual who adds the data after confirming that the organization(s) should have access to the year in question. In the cases where data is shared across multiple organizations and requires a DUA to view, this could introduce problems. Not saying this is a no, but we need to account for it!

**Possible Solutions (Minimal Marketable Features)**

**Assumptions**
Clients have already signed DUA permissions and have access to the data
Clients have sent in their data, for a particular dataset, to Stratasan

**User Story**

1. When a new year of data is received, for DUA dataset, I want the user permissions to rollover to the new year instead of having to be manually added

2. When a new year of data is added to a dataset I want only users for the org that sent the data to receive an email that the data is updated 

【 highlighted #e2bb00 】**NEXT STEPS - This ticket needs a brainstorming session with the AP Devs to figure out or talk through a new approach on DUA permissions. Once the AP Mob reconvenes in October I will set up a meeting during the Mob session and invite Brian to discuss**【 end highlighted 】

**Acceptance Criteria**

**User Steps** 

**What does success look like?** 

┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=413157490)
┆Attachments: <a href="https://share.unito.io/rtPvCdINVhMcS6yhKvgymxte0l7DR_Q7QvERB01uZgcQ">Image-2020-06-29 08:57:35.png</a> | <a href="https://share.unito.io/IJxhOkdtH-e2_z0te_31PeIX4n1lb9uMSKTw9gK9SkJL">DUA Yes.No cleanup.xlsx</a> | <a href="https://share.unito.io/e3NIkj9Ro5UOIOuI-nohsND1TvO78zHjZo85BTU8lS56">full_dataset_list_for_review.xlsx</a>

**Comment by stratabuild** [comment]
➤ Daniel Dreaden commented:

Not Urgent at this time.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Haley AttridgeAlex Antonison putting this on the agenda for next week's meeting as I believe we can handle this with better auditing of the process. Do we have the current process documented anywhere? If we are going to rely on any development, then this will not get done for a while if it requires app development.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Latest slack thread: https://stratasan.slack.com/archives/C0481V9Q6/p1593180380496800

**Comment by stratabuild** [comment]
➤ Haley Attridge commented:

I do not think any documentation currently exists around this. If it does, it might be from Brian Dailey ?

**Comment by stratabuild** [comment]
➤ Brian Dailey commented:

Our documentation around the data process in general is here:

https://docs.google.com/document/d/1bNHoq9OCKVApIL0LTAm8KvbrD6EA8VHSpTa8M1kGVrI/edit#heading=h.bsbyn51v16me

We could add a note about the DUA years, if needed.

I do not believe this is a data engineering task - this is more in the realm of the application code itself.

After reading the Slack thread, I wonder if we really need to do anything here, because it seems like removing the DUA years entirely would be sufficient.

If that is not the case, we could populate the DUA years for the length of the contract, as Dave suggested. Then, to avoid confusion, we could adjust the page in Blackbird that shows "available data" to actually show the intersection of _authorized_ and _available_ data, rather than just "authorized." This would remove the point of confusion there.

I would _not_ recommend the proposed solution because it errs on the side of granting access when I believe that administrators should _proceed with caution any time we are granting an additional year of data_. Automatically granting a year's access of data err's on the side of convenience, when providing access may not be authorized and doing so would cause us organizational harm.

Lastly, there are already tools to help with this. First, the QA page (https://app.stratasan.com/backoffice/data_approval/details/3350/) shows the years authorized for each attached company, so it's easy for a QA person to cross-reference the years available with the years authorized. See attached image. Secondly, we have a report in Backoffice (https://app.stratasan.com/backoffice/orgs/available_data_updates) that lists all organization and data set pairs where new data is available but an organization doesn't yet have access to it. If this were reviewed regularly, this would mitigate this issue.

In short, I believe we already have the tools to deal with this.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Haley AttridgeAlex Antonison reminder we are going to talk about this as part of our data engineering call.

**Comment by stratabuild** [comment]
➤ Haley Attridge commented:

Alex AntonisonChristian Lawrence I have added a "Administrative Permissions in Stratasan Application". I am not sure the wording is perfect, but would love your review. Also attached is a list of data sets that do/do not require additional limitations (additional signatures, years, etc) after it is loaded into backoffice. 

NOT required includes all APCD, Internal Client Data, and State data provided directly by the client, to name a few. 
REQUIRED includes data sets with multiple clients sending data and having access, CMS SAF data sets, Stratasan licensed state data that required clients listed on DUA. 

All in RED- no longer are utilized and can be deleted from our platform/files. 
All in YELLOW- have not been updated in years and should be reviewed for deletion.

**Comment by stratabuild** [comment]
➤ Alex Antonison commented:

Brian Dailey To bring you up to speed, I proposed we used Haley's list as a means to write an update script to go into the app database and uncheck all of the needed data sets that have "Requires DUA" checked.  The wording on this check box has lead to it being checked when it does not need to be since it only needs to be checked in the event we need to manage a dataset with a DUA by years.  More often than not that is not the case especially with APCD

Now, in the cases where data sets do need for it to be checked, we would want to leave it checked and then Haley only has to manage a handful of states until a fix can be put in place where the app shows the intersection of available data + available dua years

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Documentation has been updated here: https://docs.google.com/document/d/1bNHoq9OCKVApIL0LTAm8KvbrD6EA8VHSpTa8M1kGVrI/edit#

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Haley Attridge do we need to train CSM's and Onboarding on this new process?

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

we did some work on this but by reducing the number of impacted data sets, but this one may still require a change. For THA, there are three that still require a change to accommodate them.   
THA (Compass) Emergency	 
THA (Compass) Inpatient	 
THA (Compass) Outpatient	 
See Column A in the attached of the changes that were implemented.

**Comment by stratabuild** [comment]
➤ Daniel Dreaden commented:

When I
am working with clients who have data sets in-app

I want
the Data Services Team process to include a step where when the year kicks over into a new year, they either automatically or otherwise go ahead and update the years to which the clients have access in Admin

So I
can avoid lapses in coverage that aren't the fault of the client.

Feature request for
Data

Add screenshots or other files to help

Links to queries/reports

Internal or External
Internal

If External, please name client and org

Further Context
In Admin, data set permissions are set up with years associated. When new data sets are added to clients, the person adding must select which years are applicable. This part of the process works well (as far as I know) because it gives us a specific time to check for DUA requirements, permissions, etc.

Where there is opportunity is when we move over into a new year. Currently, it doesn't appear that there is a process in place to ensure we update the data set permissions to reflect the new year of data. This means while we may have THA (Compass) IP in house for 2019Q1 and Back Office appears to have everything good for users to have it, it is not available in the application because it has not been updated in Admin. Users get the email that new data is available, but they can't get to it in the platform. This causes some clunkiness and some lapse in proper coverage.

**Comment by stratabuild** [comment]
➤ Brian Dailey commented:

On this we really want to weigh the risk of unintentionally sharing data where we shouldn't. The reason that this process is onerous is that we want a manual sign off and a directly responsible individual who adds the data after confirming that the organization(s) should have access to the year in question. In the cases where data is shared across multiple organizations and requires a DUA to view, this could introduce problems. Not saying this is a no, but we need to account for it!
